contracts/PodCastNFT.sol

```diff
@@ -5,53 +5,60 @@ import "hardhat/console.sol";
 import "@openzeppelin/contracts/token/ERC721/extensions/ERC721URIStorage.sol";
 import "@openzeppelin/contracts/access/Ownable.sol";
 import "@openzeppelin/contracts/utils/Counters.sol";
-import "./ConsensusAdminable.sol";
 
 import {Base64} from "./libraries/Base64.sol";
 
-contract PodCastNFT is ERC721URIStorage, Ownable, ConsensusAdminable {
+contract PodCastNFT is ERC721URIStorage, Ownable {
     using Counters for Counters.Counter;
     Counters.Counter private _tokenIds;
 
-    constructor(address[] memory _admin, bool givenHighestAuthority)
-        ConsensusAdminable(_admin, givenHighestAuthority)
-        ERC721("Henkaku v0.1", "henkaku")
-    {}
+    constructor() ERC721("Henkaku v0.2", "henkaku") {}
 
+    mapping(address => string[]) public roles;
     mapping(uint256 => bool) private _communityMemberShip;
 
-    function isCommunityMember(uint256 _tokenId) public view returns (bool) {
-        return _communityMemberShip[_tokenId];
+    function getRoles(address _address) public view returns (string[] memory) {
+        require(balanceOf(_address) != 0, "wallet must have membership nft");
```

Comment:
Do you need this check given you are not giving any roles without membership nft? It's usually not useful to use `require` in the view functions

Reply:
If you want to doubly careful, you could just return false, no need to throw an exception as client has to catch the error